src/components/body/body.component.ts

```diff
@@ -125,6 +125,7 @@ export class DataTableBodyComponent implements OnInit, OnDestroy {
   @Input() set rows(val: any[]) {
     this._rows = val;
     this.rowExpansions.clear();
+    this._rowCount = this._rows.length;
```

Comment:
This won't work right since the row count varies. There is a function to get this information.

Reply:
Aah, I see that /#server-paging is indeed broken with my change. Wonder if it's worth it to maybe add more e2e tests that could potentially catch such errors.